src/msg/Connection.h

```diff
@@ -196,10 +200,8 @@ struct Connection : public RefCountedObject {
   const entity_addrvec_t& get_peer_addrs() const {
     return *peer_addrs;
   }
-  void set_peer_addr(const entity_addr_t& a) {
```

Comment:
I don't think you need to change the definitions of `set_peer_addr` and `set_peer_addrs` because a loopback connection is only created in the AsyncMessenger constructor.

Reply:
are you mean if "async_msgr->get_myaddrs() == get_peer_addrs()", this == local_connection.get()?

Reply:
see https://github.com/ceph/ceph/pull/26923/files#diff-6c0fe972daf44fc4b14e97d9c813b395R296.  that is all you need.. don't bother with any of the myaddrs changes or checks at all